impeller/renderer/backend/vulkan/test/mock_vulkan.cc

```diff
@@ -16,16 +18,34 @@ struct MockCommandBuffer {
   std::shared_ptr<std::vector<std::string>> called_functions_;
 };
 
-struct MockDevice {
-  MockDevice() : called_functions_(new std::vector<std::string>()) {}
+class MockDevice final {
+ public:
+  explicit MockDevice() : called_functions_(new std::vector<std::string>()) {}
+
   MockCommandBuffer* NewCommandBuffer() {
-    std::unique_ptr<MockCommandBuffer> buffer =
-        std::make_unique<MockCommandBuffer>(called_functions_);
+    auto buffer = std::make_unique<MockCommandBuffer>(called_functions_);
     MockCommandBuffer* result = buffer.get();
+    Lock lock(command_buffers_mutex_);
     command_buffers_.emplace_back(std::move(buffer));
     return result;
   }
+
+  std::shared_ptr<std::vector<std::string>>& GetCalledFunctions() {
+    return called_functions_;
+  }
+
+  void was_called(const std::string& function) {
+    Lock lock(called_functions_mutex_);
+    called_functions_->push_back(function);
+  }
+
+ private:
+  FML_DISALLOW_COPY_AND_ASSIGN(MockDevice);
+
+  Mutex called_functions_mutex_;
```

Comment:
Can we use `IPLR_GUARDED_BY` with these instance variables?  I'm not sure if it will work with that Mutex.

Reply:
Yup. Done.